lib/index.js

```diff
@@ -111,35 +129,70 @@ const downloadFromInfoCallback = (stream, info, options) => {
       });
     }
 
-    req = miniget(format.url, requestOptions);
-
-    req.on('response', res => {
-      if (stream._isDestroyed) { return; }
-      if (!contentLength) {
-        contentLength = parseInt(res.headers['content-length'], 10);
-      }
-    });
-    req.on('data', ondata);
+    const shouldBeChunked = !format.audioBitrate || !format.qualityLabel;
```

Comment:
use `format.hasAudio` and `format.hasVideo`. should also check that we're not chunking `format.isHLS` and `format.isDashMPD` since those are already segmented

Reply:
shouldn't the first part of the `if` statement handle `format.isHLS` and `format.isDashMPD`?

Reply:
oooh, good point!